### PR TITLE
Some improvements to multi platform support

### PR DESCRIPTION
**Pull request by m-seldin**

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -273,24 +274,28 @@ var HubotGenerator = yeoman.generators.Base.extend({
...
         this.integrationsScripts = [];
-
+        var thiz = this;
         //TODO: eliminate duplicate code
         for(var i=0;i<integrationsStrArray.length;i++){
           var element = integrationsStrArray[i];
-          if (element.indexOf("https:") < 0 && element.indexOf("ssh:") < 0) {
-            this.integrationsScripts.push("hubot-" + element);
-            this.integrations.push("hubot-" + element);
+          if(element=="") continue;
```
Comment (rqc) [inline generators/app/index.js]: - [ ] @m-seldin could this be made `===` ?

File: generators/app/index.js [hunk 2]
```diff
@@ -273,24 +274,28 @@ var HubotGenerator = yeoman.generators.Base.extend({
...
+
+          if (!element.includes("https:") && !element.includes("ssh:")) {
+            thiz.integrationsScripts.push(path
```
Comment (rqc) [inline generators/app/index.js]: - [ ] @m-seldin could we declare `integrationsScripts` and `integrations` as vars in the the scope before the `for` loop. Then remove the `thiz = this`. Then reference them without `thiz`. And after the loop is done, you may assign them to `this.integrationsScripts = integrationScripts;` and `this.integrations = integrations`. Would that work?

**Comment by rqc** [comment]
@m-seldin I am merging this. If you could please provide answers to some of the comments it would be great. Thanks for the contribution!

File: generators/app/index.js [hunk 3]
```diff
@@ -273,24 +274,28 @@ var HubotGenerator = yeoman.generators.Base.extend({
         if(integrationsStr==null)
           done();
 
+
         var integrationsStrArray = integrationsStr.split(",");
         this.integrations = [];
         this.integrationsScripts = [];
-
+        var thiz = this;
```
Comment (m-seldin) [inline generators/app/index.js]: fixed

File: generators/app/index.js (hunk 1 above)
Comment (m-seldin) [inline generators/app/index.js]: fixed

File: generators/app/index.js (hunk 2 above)
Comment (m-seldin) [inline generators/app/index.js]: this - fixed.

**Comment by m-seldin** [comment]
created a new pull request (https://github.com/eedevops/generator-hubot-enterprise/pull/25), following our talk , please merge to master.

**Pull request merged**